tests/cover/test_renaming.py

```diff
@@ -0,0 +1,30 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+from hypothesis.internal.renaming import renamed_arguments
+
+
+def test_can_rename_arguments_in_a_function_with_no_docstring():
+    @renamed_arguments(old_arg='new_arg')
+    def f(old_arg, new_arg):
```

Comment:
It doesn't really matter, but this is opposite order of our actual usage now.

Reply:
Blergh, apparently I cannot word. Better now?

Reply:
No, you had the decorator the right way around the first time. The difference is the order of arguments in the function definition.

Reply:
I think clearly I've designed a confusing interface that needs to be better documented about how to use it TBH... but that's not your job.

Reply:
I was copying `https://github.com/HypothesisWorks/hypothesis-python/pull/823#discussion_r135607068`, with a = old_arg and b = new_arg.

Reply:
Sorry, I should have used your names when writing those examples.

The expected usage pattern is that you go from:

```python
def f(old_arg):
    pass
```

to 

```python
@renamed_arguments(old_arg='new_arg')
def f(new_arg, old_arg):
    pass
```

This way people passing the argument positionally won't get the deprecation warning - it will continue to do the right thing - while people passing it as a keyword argument will.